finality-aleph/src/network/session/manager.rs

```diff
@@ -308,20 +324,29 @@ impl<NI: NetworkIdentity, D: Data> Manager<NI, D> {
         message: DiscoveryMessage<NI::AddressingInformation>,
     ) -> ManagerActions<NI::AddressingInformation> {
         let session_id = message.session_id();
+        let creator = message.0.creator();
         match self.sessions.get_mut(&session_id) {
             Some(Session {
                 handler, discovery, ..
             }) => {
                 let (maybe_address, maybe_message) =
                     discovery.handle_authentication(message, handler);
-                let maybe_command = match (maybe_address, handler.is_validator()) {
-                    (Some(address), true) => {
+                let mut maybe_command = None;
+                if let Some(address) = maybe_address {
```

Comment:
is this correct? Wont it catch some nodes that do not have a validator role?

Reply:
In my understanding `handler.is_validator()` is with respect to self, so we should ignore it. And `discovery.handle_authentication` returns `Some(address)` iff an authentication has been validated successfully, and its author is a validator (and we are not that validator) in a given session, so I think this is correct.